### PR TITLE
Create models, connect Alembic

### PR DESCRIPTION
**Pull request by KyryloKireiev**

### CODE REVIEW

File: api_blog/views.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from aiohttp import web
+
+from .models import Session
+
+
+async def index(request):
+    async with Session() as session:
+        async with session.begin():
+            qs = sa.select(1)
+            print(await session.execute(qs))
```
Comment (y-u-n-ii-a) [inline api_blog/views.py]: You don't have to print anything. Just make a request

File: api_blog/app.py [hunk 2]
```diff
@@ -24,4 +26,13 @@ async def create_app(config):
     aioreloader.start()
     app["config"] = config
     setup_routes(app)
+    app.cleanup_ctx.append(pg_context)
     return app
+
+
+async def pg_context(app):
+    url = app["config"]["db_url"]
```
Comment (y-u-n-ii-a) [inline api_blog/app.py:34]: I don't see this var in config.yaml, it's only in your local config

Comment (y-u-n-ii-a) [reply api_blog/app.py:34]: db_url

File: migrations/env.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+from logging.config import fileConfig
+
+from alembic import context
+from sqlalchemy import engine_from_config, pool
+
+from api_blog.app import get_config
+from api_blog.models import Base
+
+ALEMBIC_URL = get_config()["alembic_url"]
```
Comment (y-u-n-ii-a) [inline migrations/env.py]: it names `db_url`, isn't it?

File: api_blog/app.py (hunk 2 above)
Comment (KyryloKireiev) [reply api_blog/app.py:34]: Yes, this var in local.yaml, it should be different?

File: migrations/env.py (hunk 3 above)
Comment (KyryloKireiev) [reply migrations/env.py]: Those two urls have little different, so, I make two vars:
db_url: postgresql+asyncpg://kyrylo:password@localhost:5432/aiohttp_blog
alembic_url: postgresql://kyrylo:password@localhost/aiohttp_blog
First for sqlalchemy, second for alembic

File: migrations/env.py [hunk 4]
```diff
@@ -82,3 +90,5 @@ def run_migrations_online() -> None:
     run_migrations_offline()
 else:
     run_migrations_online()
+
+"""
```
Comment (y-u-n-ii-a) [inline migrations/env.py]: Commented out code

File: migrations/env.py [hunk 5]
```diff
@@ -25,27 +26,39 @@
...
-
 # target_metadata = None
 
 # other values from the config, defined by the needs of env.py,
 # can be acquired:
-# my_important_option = config.get_main_option("my_important_option")
+my_important_option = config.get_main_option("my_important_option")
 # ... etc.
 
+engine = create_async_engine(url=ALEMBIC_URL, future=True, echo=True)
```
Comment (y-u-n-ii-a) [inline migrations/env.py]: Better to use here `get_main_option`

Comment (y-u-n-ii-a) [reply migrations/env.py]: Instead of ALEMBIC_URL

File: migrations/env.py [hunk 6]
```diff
@@ -25,27 +26,39 @@
 
 target_metadata = Base.metadata
 
-
 # target_metadata = None
 
 # other values from the config, defined by the needs of env.py,
 # can be acquired:
-# my_important_option = config.get_main_option("my_important_option")
+my_important_option = config.get_main_option("my_important_option")
```
Comment (y-u-n-ii-a) [inline migrations/env.py]: Why did you uncomment this line? Do you need it?

File: api_blog/views.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+from aiohttp import web
+
+from .models import Session
+
+
+async def index(request):
+    async with Session() as session:
+        async with session.begin():
+            qs = sa.select(1)
+            await session.execute(qs)
```
Comment (y-u-n-ii-a) [inline api_blog/views.py]: Let's refactor it a little bit)
```suggestion
            await session.execute(sa.select(1))
```

File: migrations/env.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
+import asyncio
+from logging.config import fileConfig
+
+from alembic import context
+from sqlalchemy.ext.asyncio import create_async_engine
+
+from api_blog.app import get_config
+from api_blog.models import Base
+
+ALEMBIC_URL = get_config()["db_url"]
```
Comment (y-u-n-ii-a) [inline migrations/env.py]: ALEMBIC_URL - let's get rid of this var

File: migrations/env.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+target_metadata = Base.metadata
+
+# target_metadata = None
+
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+engine = create_async_engine(url=ALEMBIC_URL, future=True, echo=True)
```
Comment (y-u-n-ii-a) [inline migrations/env.py]: Please do not create the engine globally. Create it instead of your function

File: api_blog/models.py [hunk 10]
```diff
@@ -25,9 +25,18 @@ class Article(BaseModel):
...
 
     title = Column(String(100), nullable=False)
+
+
+class User(BaseModel):
+    __tablename__ = "user"
+
+    username = Column(String(50), nullable=False, unique=True)
+    email = Column(String(100), nullable=False, unique=True)
+    password_hash = Column(String(128), nullable=False)
```
Comment (y-u-n-ii-a) [inline api_blog/models.py]: Just `password`

File: api_blog/models.py [hunk 11]
```diff
@@ -25,9 +25,18 @@ class Article(BaseModel):
...
     __tablename__ = "category"
 
     title = Column(String(100), nullable=False)
+
+
+class User(BaseModel):
+    __tablename__ = "user"
+
+    username = Column(String(50), nullable=False, unique=True)
+    email = Column(String(100), nullable=False, unique=True)
```
Comment (y-u-n-ii-a) [inline api_blog/models.py]: Let's use just username

**Review by y-u-n-ii-a — APPROVED** [review]

**Pull request merged**